crates/server/src/builder.rs

```diff
@@ -96,7 +99,9 @@ impl<TJsonRpcHandler: RpcHandler, THttpApiHandler: Clone + Send + Sync + 'static
             svc = svc.layer(Extension(self.http_api_handler.unwrap()));
         }
 
-        svc = svc.layer(TraceLayer::new_for_http());
+        svc = svc
+            .layer(TraceLayer::new_for_http())
+            .layer(TimeoutLayer::new(Duration::from_secs(starknet_config.timeout.into())));
```

Comment:
Why do we need this layer?

Reply:
to set a timeout, I discussed that `TimeoutLayer` with @marioiordanov

Reply:
What this timeout used for?

Reply:
in devnet in python, sometimes execution was longer than 30, or 60 seconds so there was a need to exceed timeout and make it a CLI parameter